src/components/APIDoc/CodeBlockDropdown.tsx

```diff
@@ -2,14 +2,13 @@ import React from 'react';
 import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
 
 import { SnippetInfoItem, SnippetItemsArray } from '../../hooks/useSnippets';
+import { useSetLanguage, useLanguage } from '../../utils/LanguageContext';
 
 
-export interface CodeBlockDropdownProps {
-  language: SnippetInfoItem;
-  setLanguage: React.Dispatch<React.SetStateAction<SnippetInfoItem>>;
-}
+export const CodeBlockDropdown: React.FunctionComponent = () => {
+  const language = useLanguage();
+  const setLanguage = useSetLanguage();
 
-export const CodeBlockDropdown: React.FunctionComponent<CodeBlockDropdownProps> = ({ language, setLanguage }) => {
   const [isOpen, setIsOpen] = React.useState(false);
   const [selected, setSelected] = React.useState(language.text);
```

Comment:
`language` is acting as an initializer for the selected language.
I suggest to remove this line altogether and change all the references of `selected` to `language.text`. 
Remove references to `setSelected`.

You are basically having 2 sources of truth here (`language` and `selected`). hence why we should get rid of `selected`